reana_client/utils.py

```diff
@@ -200,15 +205,32 @@ def _validate_environment(reana_yaml):
         workflow_steps = reana_yaml["workflow"]["specification"]["steps"]
         _validate_serial_workflow_environment(workflow_steps)
     elif workflow_type == "yadage":
-        logging.warning(
-            "The environment validation is not implemented for Yadage workflows yet."
-        )
+        workflow_steps = reana_yaml["workflow"]["specification"]["stages"]
+        _validate_yadage_workflow_environment(workflow_steps)
     elif workflow_type == "cwl":
         logging.warning(
             "The environment validation is not implemented for CWL workflows yet."
         )
 
 
+def _validate_yadage_workflow_environment(workflow_steps):
+    """Validate environments in REANA yadage workflow.
+
+    :param workflow_steps: List of dictionaries which represents different steps involved in workflow.
+    :raises Warning: Warns user if the workflow environment is invalid in yadage workflow steps.
+    """
+    for stage in workflow_steps:
+        environment = stage["scheduler"]["step"]["environment"]
+        image = "{}:{}".format(environment["image"], environment["imagetag"])
+        image_name, image_tag = _validate_image_tag(image)
+        _image_exists(image_name, image_tag)
+        uid, gids = _get_image_uid_gids(image_name, image_tag)
+        k8s_uid = None
+        if environment.get("resources"):
+            k8s_uid = environment.get("resources")[0].get("kubernetes_uid")
```

Comment:
This won't work if the `kubernetes_uid` is not in the first position, [which is perfectly possible](https://github.com/yadage/yadage-schemas/blob/260fbc40ec5b8afa7d59c0dd00591e02e13774e4/yadageschemas/packtivity/environment/docker-enc-schema.json#L22):
```python
In [2]: environment
Out[2]: 
{'environment_type': 'docker-encapsulated',
 'image': 'reanahub/reana-env-root6',
 'imagetag': '6.18.04',
 'resources': [{'compute_backend': 'htcondorcern'},
  {'htcondor_max_runtime': 'espresso'}]}

In [3]: environment['resources']
Out[3]: [{'compute_backend': 'htcondorcern'}, {'htcondor_max_runtime': 'espresso'}]

In [4]: environment['resources'].append({'kubernetes_uid': '500'})

In [5]: environment['resources']
Out[5]: 
[{'compute_backend': 'htcondorcern'},
 {'htcondor_max_runtime': 'espresso'},
 {'kubernetes_uid': '500'}]

In [6]: environment['resources'][0].get('kubernetes_uid')

In [7]: environment['resources'][2].get('kubernetes_uid')
Out[7]: '500'
```

Reply:
Good catch! Any idea why they're using a list when it could be a simple dictionary?

I guess something like this should do...
```python
next((resource['kubernetes_uid'] for resource in resources if 'kubernetes_uid' in resource), None)
```

Reply:
Tested on `r-d-atlas-recast`.

Reply:
Have you tested it by adding more resources as in @diegodelemos's example?